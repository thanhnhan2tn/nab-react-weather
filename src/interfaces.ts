export interface IWeatherInfo {
  air_pressure?: number
  id: number
  applicable_date?: Date | string
  time?:string,
  title?: string,
  the_temp?: number,
  max_temp?: number
  min_temp?: number
  weather_state_abbr: string
  weather_state_name: string
  wind_speed?: number
}

export interface IWeather {
  title: string
  time: string
  consolidated_weather: IWeatherInfo[]
}
