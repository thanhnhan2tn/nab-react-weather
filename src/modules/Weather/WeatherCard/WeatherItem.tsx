
import React from 'react';
import moment from 'moment';

import { IWeatherInfo } from '../../../interfaces';
import { WEATHER_STATE } from '../../../constants/weather-state';

const WeatherItem = ({ data: {
  applicable_date,
  min_temp,
  max_temp,
  weather_state_abbr,
  weather_state_name,
} }: { data: IWeatherInfo }) => (
  <>
    <div className="row row1">Min: {min_temp ? Math.round(min_temp): '...'} &deg;</div>
    <div className="row row1">Max: {max_temp ? Math.round(max_temp): '...'} &deg;</div>
    <div className="row row2">
      {
        weather_state_abbr ?
          <img
            className="img-fluid"
            alt={weather_state_name}
            src={WEATHER_STATE[weather_state_abbr]}
          />
          : null
      }
    </div>
    <div className="row row3">{moment(applicable_date).format('dddd')}</div>
  </>
);


export default WeatherItem;