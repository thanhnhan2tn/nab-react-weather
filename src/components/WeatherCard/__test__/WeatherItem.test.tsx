import React from 'react';
import { shallow } from 'enzyme';
import WeatherItem from '../WeatherItem';
import { IWeatherInfo } from '../../../interfaces';

describe('WeatherItem', () => {
  let props: { data: IWeatherInfo };
  beforeEach(() => {
    props = {
      data: {
        id: 1,
        time: '2022-01-01T07:16:41.376284Z"',
        title: 'Ho Chi Minh',
        the_temp: 31,
        weather_state_abbr: 'hr',
        weather_state_name: 'Heavy Rain'
      }
    };
  });
  it('should render to match snapshot', () => {
    const component = shallow(<WeatherItem  {...props} />);
    expect(component).toMatchSnapshot();
  })
})