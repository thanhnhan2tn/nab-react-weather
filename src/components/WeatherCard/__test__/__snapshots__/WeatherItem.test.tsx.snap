// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem should render to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherItem
    data={
      Object {
        "id": 1,
        "the_temp": 31,
        "time": "2022-01-01T07:16:41.376284Z\\"",
        "title": "Ho Chi Minh",
        "weather_state_abbr": "hr",
        "weather_state_name": "Heavy Rain",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="row row1"
        >
          Min: 
          ...
           °
        </div>,
        <div
          className="row row1"
        >
          Max: 
          ...
           °
        </div>,
        <div
          className="row row2"
        >
          <img
            alt="Heavy Rain"
            className="img-fluid"
            src="https://www.metaweather.com/static/img/weather/hr.svg"
          />
        </div>,
        <div
          className="row row3"
        >
          Sunday
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Min: ",
            "...",
            " °",
          ],
          "className": "row row1",
        },
        "ref": null,
        "rendered": Array [
          "Min: ",
          "...",
          " °",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Max: ",
            "...",
            " °",
          ],
          "className": "row row1",
        },
        "ref": null,
        "rendered": Array [
          "Max: ",
          "...",
          " °",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Heavy Rain"
            className="img-fluid"
            src="https://www.metaweather.com/static/img/weather/hr.svg"
          />,
          "className": "row row2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Heavy Rain",
            "className": "img-fluid",
            "src": "https://www.metaweather.com/static/img/weather/hr.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sunday",
          "className": "row row3",
        },
        "ref": null,
        "rendered": "Sunday",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="row row1"
          >
            Min: 
            ...
             °
          </div>,
          <div
            className="row row1"
          >
            Max: 
            ...
             °
          </div>,
          <div
            className="row row2"
          >
            <img
              alt="Heavy Rain"
              className="img-fluid"
              src="https://www.metaweather.com/static/img/weather/hr.svg"
            />
          </div>,
          <div
            className="row row3"
          >
            Sunday
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Min: ",
              "...",
              " °",
            ],
            "className": "row row1",
          },
          "ref": null,
          "rendered": Array [
            "Min: ",
            "...",
            " °",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Max: ",
              "...",
              " °",
            ],
            "className": "row row1",
          },
          "ref": null,
          "rendered": Array [
            "Max: ",
            "...",
            " °",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Heavy Rain"
              className="img-fluid"
              src="https://www.metaweather.com/static/img/weather/hr.svg"
            />,
            "className": "row row2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Heavy Rain",
              "className": "img-fluid",
              "src": "https://www.metaweather.com/static/img/weather/hr.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sunday",
            "className": "row row3",
          },
          "ref": null,
          "rendered": "Sunday",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
